catalog/core/catalog-core-impl/filter-proxy/src/test/java/ddf/catalog/filter/proxy/adapter/test/FilterAdapterTest.java

```diff
@@ -624,6 +626,26 @@ public void testTemporalRelativeFilter() {
     }
   }
 
+  @Test
+  public void testParseDuration() {
+    String query = "RELATIVE(P3.14159Y3M2DT.25H6M)";
+    BigDecimal secondsPerYear = new BigDecimal(ChronoUnit.YEARS.getDuration().getSeconds());
+    BigDecimal secondsPerHour = new BigDecimal(ChronoUnit.HOURS.getDuration().getSeconds());
+    BigDecimal yearFraction = new BigDecimal(0.14159);
+    BigDecimal hourFraction = new BigDecimal(0.25);
+    int seconds =
+        yearFraction.multiply(secondsPerYear).add(hourFraction.multiply(secondsPerHour)).intValue();
+    org.joda.time.Period period = new org.joda.time.Period(3, 3, 0, 2, 0, 6, 0, 0);
+    period = period.plusSeconds(seconds);
+
+    try {
+      GeotoolsFilterAdapterImpl geotoolsFilterAdapter = new GeotoolsFilterAdapterImpl();
+      assertThat(geotoolsFilterAdapter.parsePeriod(query), is(period));
+    } catch (Exception e) {
```

Comment:
✏️ Throw a more specific exception here so that we'll be better able to diagnose the problem if this test fails.